### DISCUSSION TITLE
[PowerShell] Delta works, but the Diff seems not do be correct - How to setup correctly

### DISCUSSION THREAD

**Discussion by GregOnNet** [dandavison/delta#1915] [Q&A] [answered]
Hi,

I think I miss something about setting up delta properly on Windows with PowerShell.
The diff is produced, but it looks a bit weird.
Or maybe, I have the wrong expectations.

When I execute `git diff` the diff shows the changes.
However, line 1 is shown twice.
I would have thought that line 1 is showing once, indicating that the name has been changed.
Currently, it looks like that one line has been removed and a new one was created.

![image](https://github.com/user-attachments/assets/20b5d23a-7cfa-4d88-89d7-c3e3ea0625d8)

## Here is how I setup delta

```bash
winget install dandavison.delta
winget install jftuga.less
```

## This less-version is exposed via PATH
> Seems to be correct

```bash
less --version
less 668 (Spencer V8 regular expressions)
Copyright (C) 1984-2024  Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Home page: https://greenwoodsoftware.com/less
```

## Here is how I configured delta in git

```
[core]
  pager = delta
  whitespace = cr-at-eol
  quotepath = false
  editor = code --wait
  eol = lf
  autocrlf = input

[interactive]
    diffFilter = delta --color-only --features=interactive

[delta]
    hyperlinks = true
    hyperlinks-file-link-format = "vscode://file/{path}:{line}"
    features = zebra-dark line-numbers decorations
    syntax-theme = Sublime Snazzy
    plus-style = syntax "#003800"
    minus-style = syntax "#3f0001"

[delta "decorations"]
    commit-decoration-style = bold yellow box ul
    file-style = bold yellow ul
    file-decoration-style = none
    hunk-header-decoration-style = cyan box ul

[delta "line-numbers"]
    line-numbers-left-style = cyan
    line-numbers-right-style = cyan
    line-numbers-minus-style = 124
    line-numbers-plus-style = 28
```

I am glad for any hint. :-)

**Comment by GregOnNet** [comment]
I also checked with a TypeScript-File, because I thought, maybe it helps delta if more context like Syntax is present.
I see the same behaviour.
Syntax highlighting seems to to work either.
I activated `diff-highlight` in `.gitconfig`, btw.

```diff
[delta]
  navigate = true
  hyperlinks = true
  hyperlinks-file-link-format = "vscode://file/{path}:{line}"
- features = zebra-dark unobtrusive-line-numbers decorations
+ features = zebra-dark diff-highlight unobtrusive-line-numbers decorations
```

![image](https://github.com/user-attachments/assets/d2cb9226-fa9b-457f-994b-26cfef76c97d)

**Comment by dandavison** [comment, marked as answer]
Hi @GregOnNet, that all looks correct. You get a "minus" line and a "plus" line, just like with normal git diff. For syntax highlighting you need to remove diff-highlight. In fact I'd recommend trying with minimal settings to start off with and then only adding more if you need them.

**Comment by GregOnNet** [comment]
Thanks @dandavison ,

yeah my expectation was incorrect.
I got confused with `--word-diff` output.

I tried once more with more changes and now I also get the nice decoration, in which Method the changes has been made.
I also simplified my configuration as you suggested.

![image](https://github.com/user-attachments/assets/edd05347-adfd-4a21-8ab0-6f282d40c180)

**Reply by dandavison** [reply]
Great!
